File: SVR.R
# SVR

#Importing the dataset
dataset = read.csv('Position_Salaries.csv')
dataset = dataset[2:3]

#Splitting the dataset
#install.packages('caTools')
#library(caTools)
#set.seed(123)
#split = sample.split(dataset$, SplitRatio = 2/3)
#training_set = subset(dataset, split = TRUE)
#test_set = subset(dataset, split = FALSE)

#Feature Scalling
#dataset = scale(dataset)

# Fitting the SVR model to the dataset
#install.packages('e1071')
library(e1071)
regressor = svm(formula=Salary ~ .,
                data=dataset,
                type='eps-regression')

# Predicting a new result with SVR
y_pred = predict(regressor, data.frame(Level = 6.5))

# Visualising the SVR result
library(ggplot2)
ggplot() +
  geom_point(aes(x = dataset$Level, y = dataset$Salary),
             color = 'red') + 
  geom_line(aes(x = dataset$Level, y = predict(regressor, newdata = dataset)),
            color = 'blue') +
  ggtitle('Truth or Bluff(SVR Model)') +
  xlab('Level') + 
  ylab('Salary')

# Visualising the SVR result (smoother)
library(ggplot2)
x_grid = seq(min(dataset$Level), max(dataset$Level), 0.1)
ggplot() +
  geom_point(aes(x = dataset$Level, y = dataset$Salary),
             color = 'red') + 
  geom_line(aes(x = x_grid, y = predict(regressor, newdata = data.frame(Level = x_grid))),
            color = 'blue') +
  ggtitle('Truth or Bluff(SVR Model)') +
  xlab('Level') + 
  ylab('Salary')

File: SVR.py
# -*- coding: utf-8 -*-
"""
Created on Wed Jan  2 11:43:18 2019

@author: LENOVO
"""
# SVR 

# Importing Libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importing Dataset
dataset = pd.read_csv('Position_Salaries.csv')
X =  dataset.iloc[:, 1:2].values
#X = pd.DataFrame(Xnd)
y = dataset.iloc[:, 2:3].values
#y = pd.DataFrame(ynd)

# Splitting the dataset into the Training set and Test set
"""from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.2, 
                                                    random_state = 0)"""

# Feature Scaling
from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
sc_y = StandardScaler()
X = sc_X.fit_transform(X)
y = sc_y.fit_transform(y)

# Fitting the SVR Model to the dataset
from sklearn.svm import SVR
regressor = SVR(kernel = 'rbf')
regressor.fit(X, y)

# Predicting a new result with SVR
y_pred = sc_y.inverse_transform(regressor.predict(sc_X.transform(np.array([[6.5]]))))

# Visualising the SVR Results 
plt.scatter(X,y, color = 'red')
plt.plot(X, regressor.predict(X), color = 'blue')
plt.title('Truth or Bluff (SVR model)')
plt.xlabel('Position Level')
plt.ylabel('Salary')
plt.show

# Visualising the SVR Results (higher resolution)
X_grid = np.arange(min(X), max(X),0.1)
X_grid = X_grid.reshape(len(X_grid), 1)
plt.scatter(X,y, color = 'red')
plt.plot(X_grid, regressor.predict(X_grid), color = 'blue')
plt.title('Truth or Bluff (SVR model)')
plt.xlabel('Position Level')
plt.ylabel('Salary')
plt.show
